src/posix/platform/radio.cpp

```diff
@@ -89,6 +93,22 @@ void otPlatRadioSetPromiscuous(otInstance *aInstance, bool aEnable)
 
 void platformRadioInit(const otPlatformConfig *aPlatformConfig)
 {
+#if OPENTHREAD_POSIX_CONFIG_MAX_POWER_TABLE_ENABLE
+    if (aPlatformConfig->mMaxPowerTable != NULL)
```

Comment:
How about moving the parsing string code to the class `MaxPowerTable` as a function?